docs/README-Extending.md

```diff
@@ -0,0 +1,136 @@
+```
+Notice the syntax we use is for the context value or viewItem above is a regular expression as denoted by the "=~" equal test. Using regular expressions 
+to describe context allows more meaning to be embedded in the context.
+
+Additional contexts are being indicated by the use of a preceding underscore character. These need to be added in the Zowe Explorer so the process to get them added 
+is by raising a [git issue](https://github.com/zowe/vscode-extension-for-zowe/issues) and requesting a specific hook.
+
+
+## Zowe Explorer Extender API
+
+The following code snippet shows how to gain access to the published Zowe Explorer extender API.
+
+```javascript
+export function activate(context: vscode.ExtensionContext) {
+    const EXTENDER_API;
+    const baseExt = vscode.extensions.getExtension('zowe.vscode-extension-for-zowe');
+    try {
+        if (baseExt && baseExt.exports) {
+            EXTENDER_API = baseExt.exports.getExplorerExtenderApi();
+            return;
+        }
+    } catch (error) {
+        vscode.window.showWarningMessage("Unable to access Zowe Explorer API");
+    }
+}
+```
+
+Details of function available in the Api can be found by looking in the [Zowe Explorer source](https://github.com/zowe/vscode-extension-for-zowe/blob/master/src/api/ZoweExplorerExtender.ts) 
+Initially the following access methods were added to provide access to the appropriate profiles.
+
+```javascript
+        /**
+         * Used by other VS Code Extensions to access the primary profile.
+         *
+         * @param primaryNode represents the Tree item that is being used
+         * @return The requested profile
+         *
+         */
+        getProfile(primaryNode: IZoweTreeNode): IProfileLoaded;
+        
+        /**
+         * Used by other VS Code Extensions to access an alternative
+         * profile types that can be employed in conjunction with the primary
+         * profile to provide alternative support.
+         *
+         * @param primaryNode represents the Tree item that is being used
+         * @return The requested profile
+         */
+        getLinkedProfile(primaryNode: IZoweTreeNode, type: string): Promise<IProfileLoaded>;
+```
+
+## The relationship with CLI profiles
+
+Zowe Explorer uses the Zowe CLI profiles that have been created by the user and accesses information via a primary profile such as z/OSMF
+
+!["Primary Profile"](./images/ZE-basic.gif?raw=true "Primary Profile")
+
+This diagram shows that information presented in the Zowe Explorer has been retrieved from a z/OS server via an instance of a z/OSMF profile. Because
+Zowe Explorer embeds the Zowe CLI node modules it uses this profile in conjunction with the API layer of the CLI plugin to create the
+displayed content and provide functionality such as menus and other options.
+
+There are of course other sources of information that could be presented that is not available via the primary profile. For example specific vendor tools associated with
```

Comment:
There are of course other sources of information that could be presented that are not available via the primary profile. Examples include specific vendor tools associated with Jobs, or functions that represent advanced file management.